fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/EquivalenceClasses.java

```diff
@@ -0,0 +1,54 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.optimizer.base;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class EquivalenceClasses {
+    private final Map<ColumnRefOperator, Set<ColumnRefOperator>> columnToEquivalenceClass;
+
+    public EquivalenceClasses() {
+        columnToEquivalenceClass = Maps.newHashMap();
+    }
+
+    public void addEquivalence(ColumnRefOperator left, ColumnRefOperator right) {
```

Comment:
use ScalarEquivalenceExtractor more eazy? you can add the interface in ScalarEquivalenceExtractor, and only need put `left = right` then you work is over

Reply:
I want all column ref in the same equivalence class has the same ordered equivalence. So I add this class.
It seems that ScalarEquivalenceExtractor can not achieve this goal?

Reply:
I'm not sure, I just think it's repetitive work, and in the future the class may become the same as ScalarEquivalenceExtractor

Reply:
I review the logic of ScalarEquivalenceExtractor. It does not realize my requirements.
here I just keep this class. If necessary I will merge it with ScalarEquivalenceExtractor later.